### PR TITLE
trackEvent mutation and tabs

### PR DESCRIPTION
**Pull request by rileylnapier**
## Description

[Pull Request Description Here]

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-inbox/src/components/OptionsDropdown/Options.tsx [hunk 1]
```diff
@@ -22,23 +22,25 @@ const Option = styled.div`
...
   return (
     <Container>
-      {
-        options.map(({ label, onClick = () => {} }) => <Option key={label} onClick={() => {
-          onClick();
-          close()
-        }}>{label}</Option>)
-      }
+      {options.map(({ label, onClick = () => {} }) => (
+        <Option key={label} onClick={onClick}>
```
Comment (rileylnapier) [inline packages/react-inbox/src/components/OptionsDropdown/Options.tsx:38]: don't onclick a div

File: packages/react-inbox/src/components/OptionsDropdown/index.tsx [hunk 2]
```diff
@@ -29,24 +29,22 @@ const StyledTippy = styled(Tippy)`
...
-  offset: [0, -8]
+  offset: [0, -8],
 };
 
 function OptionsDropdown({ options }) {
-  const [showDropdown, setShowDropdown] = useState(false);
-  
   return (
     <Container>
-      <StyledTippy {...tippyProps} visible={showDropdown} content={<Options close={() => setShowDropdown(false)} options={options} />}>
```
Comment (rileylnapier) [inline packages/react-inbox/src/components/OptionsDropdown/index.tsx:43]: tippy was getting mad by having "visible" and "trigger" props at the same time

File: packages/react-inbox/src/hooks/use-inbox.ts [hunk 3]
```diff
@@ -1,47 +1,144 @@
...
+    transport?.listen({
+      id: "inbox-listener",
+      listener: (courierEvent) => {
+        newMessage(courierEvent?.data);
+      },
+    });
+  }, [transport]);
 
   return {
     ...inbox,
```
Comment (RoyalSix) [inline packages/react-inbox/src/hooks/use-inbox.ts:37]: Would like to see how we can get a more declarative pattern for this, I keep getting confused about whats actually in the `inbox` object. (Don't need to solve this here)

File: packages/react-inbox/src/hooks/use-inbox.ts [hunk 4]
```diff
@@ -1,47 +1,144 @@
...
-        type: "inbox/NEW_MESSAGE",
-        payload,
+        type: "inbox/SET_ALL_MESSAGES",
+        payload
+      });
+    },
+
+    markMessageRead: async (messageId: string, trackingId: string) => {
+      await trackEvent({
+        trackingId
```
Comment (RoyalSix) [inline packages/react-inbox/src/hooks/use-inbox.ts:118]: Nice abstraction 👍🏾

File: packages/react-inbox/src/hooks/use-inbox.ts [hunk 5]
```diff
@@ -1,47 +1,144 @@
...
       });
     },
+
+    markMessageUnread: async (messageId: string, trackingId: string) => {
+      await trackEvent({
+        trackingId
+      });
+
+      dispatch({
+        type: "inbox/MARK_MESSAGE_UNREAD",
```
Comment (RoyalSix) [inline packages/react-inbox/src/hooks/use-inbox.ts:135]: Would like to have this as a constant namespace in the future

**Review by RoyalSix — DISMISSED** [review]

File: packages/react-inbox/src/hooks/use-inbox.ts (hunk 3 above)
Comment (rileylnapier) [reply packages/react-inbox/src/hooks/use-inbox.ts:28]: i feel the same, right now its basically the main props passed into the component

File: packages/react-inbox/src/hooks/use-inbox.ts (hunk 5 above)
Comment (rileylnapier) [reply packages/react-inbox/src/hooks/use-inbox.ts:97]: agree

File: packages/react-provider/src/hooks/use-graphql-client.ts [hunk 6]
```diff
@@ -8,6 +8,7 @@ export default () => {
   const client = useMemo(() => {
     return createClient({
       url: `${apiUrl ?? process.env.API_URL ?? `https://api.courier.com`}/client/q`,
+      requestPolicy: "network-only",
```
Comment (rileylnapier) [inline packages/react-provider/src/hooks/use-graphql-client.ts:11]: this will need to change

File: packages/react-inbox/src/reducer.ts [hunk 7]
```diff
@@ -70,10 +70,13 @@ export default (state: InboxState = {}, action) => {
   }
 
   case "inbox/MARK_MESSAGE_READ": {
+    const unreadMessageCount = (state.unreadMessageCount ?? 1) - 1;
```
Comment (RoyalSix) [inline packages/react-inbox/src/reducer.ts:73]: Not sure I understand why its -1 but approving. Maybe we could get a `Math.max(unreadMessageCount, 0)`

**Review by RoyalSix — APPROVED** [review]

File: packages/react-inbox/src/reducer.ts (hunk 7 above)
Comment (rileylnapier) [reply packages/react-inbox/src/reducer.ts:73]: This is a good idea

**Pull request merged**